tests/core_tests/state_test.py

```diff
@@ -0,0 +1,128 @@
+import unittest, logging
+from core.state import State
+
+import numpy as np
+
+DEBUG = False
+
+
+class StateTestCases(unittest.TestCase):
+    def test_state_fields(self):
+        s_0 = {
+            "time": 0.0,
+            "ang_vel_x": 0.0,
+            "ang_vel_y": 0.0,
+            "ang_vel_z": 0.0,
+            "gnc_pos_q1": 0.0,
+            "gnc_pos_q2": 0.0,
+            "gnc_pos_q3": 0.0,
+            "gnc_pos_q4": 0.0,
+            "vel_x": 0.0,
+            "vel_y": 0.0,
+            "vel_z": 0.0,
+            "x": 0.0,
+            "y": 0.0,
+            "z": 0.0,
+            "force_propulsion_thrusters": 0.0,
+            "fuel_mass": 0.0,
+            "force_earth": 0.0,
+            "force_moon": 0.0,
+            "propulsion_on": False,
+            "solenoid_actuation_on": False,
+        }
+
+        bool_fields = ["propulsion_on", "solenoid_actuation_on"]
+
+        for field in s_0.keys():
+            dummy_data = 1.0
+            if field in bool_fields:
+                dummy_data = True
+            s_copy = dict(s_0)
+            s_copy[field] = dummy_data
+            self.assertEqual(
+                s_copy,
+                State({field: dummy_data}).__dict__,
+            )
+
+        self.assertEqual(
+            {
+                "time": 0.0,
+                "ang_vel_x": 0.0,
+                "ang_vel_y": 0.0,
+                "ang_vel_z": 0.0,
+                "gnc_pos_q1": 0.0,
+                "gnc_pos_q2": 0.0,
+                "gnc_pos_q3": 0.0,
+                "gnc_pos_q4": 0.0,
+                "vel_x": 0.0,
+                "vel_y": 0.0,
+                "vel_z": 0.0,
+                "x": 0.0,
+                "y": 0.0,
+                "z": 0.0,
+                "force_propulsion_thrusters": 0.0,
+                "fuel_mass": 0.0,
+                "force_earth": 0.0,
+                "force_moon": 0.0,
+                "propulsion_on": False,
+                "solenoid_actuation_on": False,
+            },
```

Comment:
I'd move `s_0` to the class or file-level scope so you don't have to copy-paste it everywhere.